### PR TITLE
Added multi-codec example

### PR DESCRIPTION
**Pull request by gfronza**

### CODE REVIEW

File: src/test/java/com/bitmovin/api/examples/CreateMultiCodecEncodingWithDashHls.java [hunk 1]
```diff
@@ -0,0 +1,718 @@
...
+		Assert.assertTrue(waitUntilFinished(encoding));
+
+		List<Manifest> manifestsToStart = new ArrayList<>();
+
+		// create all manifests
+		for (ManifestType manifestType : MANIFESTS)
+		{
+			switch (manifestType)
+			{
+			case DASH: manifestsToStart.add(createDashManifest(encoding, output)); break;
```
Comment (rkersche) [inline src/test/java/com/bitmovin/api/examples/CreateMultiCodecEncodingWithDashHls.java]: case needs to be indented. manifestToStart and break go in the next line

File: src/test/java/com/bitmovin/api/examples/CreateMultiCodecEncodingWithDashHls.java [hunk 2]
```diff
@@ -0,0 +1,718 @@
...
+			case HLS: manifestsToStart.add(createHlsManifest(encoding, output)); break;
```
Comment (rkersche) [inline src/test/java/com/bitmovin/api/examples/CreateMultiCodecEncodingWithDashHls.java]: Same here

File: src/test/java/com/bitmovin/api/examples/CreateMultiCodecEncodingWithDashHls.java [hunk 3]
```diff
@@ -0,0 +1,718 @@
...
+					String path = muxing.getOutputs().get(0).getOutputPath().replaceAll(OUTPUT_BASE_PATH, "");
+					String audioGroupId = String.format("audio_%d", audioProfile.bitrate);
+
+					audioGroupIds.add(audioGroupId);
+
+					MediaInfo audioMediaInfo = new MediaInfo();
+			        audioMediaInfo.setName(audioProfile.language);
+			        audioMediaInfo.setUri(String.format("audio_%s_%d.m3u8", audioProfile.getCodecType().toString().toLowerCase(), audioProfile.bitrate));
+					audioMediaInfo.setGroupId(audioGroupId );
+			        audioMediaInfo.setType(MediaInfoType.AUDIO);
```
Comment (rkersche) [inline src/test/java/com/bitmovin/api/examples/CreateMultiCodecEncodingWithDashHls.java]: Indent

File: src/test/java/com/bitmovin/api/examples/CreateMultiCodecEncodingWithDashHls.java [hunk 4]
```diff
@@ -0,0 +1,718 @@
...
+			{
+				if (muxing.getType() == MuxingType.TS)
+				{
+					String path = muxing.getOutputs().get(0).getOutputPath().replaceAll(OUTPUT_BASE_PATH, "");
+					String audioGroupId = String.format("audio_%d", audioProfile.bitrate);
+
+					audioGroupIds.add(audioGroupId);
+
+					MediaInfo audioMediaInfo = new MediaInfo();
+			        audioMediaInfo.setName(audioProfile.language);
```
Comment (rkersche) [inline src/test/java/com/bitmovin/api/examples/CreateMultiCodecEncodingWithDashHls.java]: Indent

File: src/test/java/com/bitmovin/api/examples/CreateMultiCodecEncodingWithDashHls.java [hunk 5]
```diff
@@ -0,0 +1,718 @@
...
+			        audioMediaInfo.setIsDefault(false);
+			        audioMediaInfo.setForced(false);
+			        audioMediaInfo.setSegmentPath(path);
+			        bitmovinApi.manifest.hls.createMediaInfo(hlsManifest, audioMediaInfo);
+				}
+			}
+		}
+
+        for (String audioGroupId : audioGroupIds)
+		{
```
Comment (rkersche) [inline src/test/java/com/bitmovin/api/examples/CreateMultiCodecEncodingWithDashHls.java]: Incorrect indention

File: src/test/java/com/bitmovin/api/examples/CreateMultiCodecEncodingWithDashHls.java [hunk 6]
```diff
@@ -0,0 +1,718 @@
...
+            for (VideoProfile videoProfile : reverseList(VIDEO_ENCODING_PROFILES))
+	    		{
```
Comment (rkersche) [inline src/test/java/com/bitmovin/api/examples/CreateMultiCodecEncodingWithDashHls.java]: Incorrect indention

File: src/test/java/com/bitmovin/api/examples/CreateMultiCodecEncodingWithDashHls.java [hunk 7]
```diff
@@ -0,0 +1,718 @@
...
+	    				}
+	    			}
+	    		}
+		}
+
+		return hlsManifest;
+	}
+
+	private VideoProfile[] reverseList(VideoProfile[] videoEncodingProfiles)
+    {
```
Comment (rkersche) [inline src/test/java/com/bitmovin/api/examples/CreateMultiCodecEncodingWithDashHls.java]: Incorrect indention

File: src/test/java/com/bitmovin/api/examples/CreateMultiCodecEncodingWithDashHls.java [hunk 8]
```diff
@@ -0,0 +1,718 @@
...
+	    					        audioGroupId, path, hlsManifest);
+	    				}
+	    			}
+	    		}
+		}
+
+		return hlsManifest;
+	}
+
+	private VideoProfile[] reverseList(VideoProfile[] videoEncodingProfiles)
```
Comment (rkersche) [inline src/test/java/com/bitmovin/api/examples/CreateMultiCodecEncodingWithDashHls.java]: Please use `ArrayUtils.reverse` from 
https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/ArrayUtils.html

File: src/test/java/com/bitmovin/api/examples/CreateMultiCodecEncodingWithDashHls.java [hunk 9]
```diff
@@ -0,0 +1,718 @@
...
+				}
+			}
+		}
+
+		return dashManifest;
+	}
+
+	private void startManifestGeneration(AbstractApiResponse manifest)
+            throws BitmovinApiException, IOException, RestException, URISyntaxException, UnirestException
+    {
```
Comment (rkersche) [inline src/test/java/com/bitmovin/api/examples/CreateMultiCodecEncodingWithDashHls.java]: Incorrect indention

File: src/test/java/com/bitmovin/api/examples/CreateMultiCodecEncodingWithDashHls.java [hunk 10]
```diff
@@ -0,0 +1,718 @@
...
+	private FMP4Muxing createFMP4Muxing(Encoding encoding, Output output, String path, Stream stream)
+			throws BitmovinApiException, IOException, RestException, URISyntaxException, UnirestException
+	{
+		EncodingOutput encodingOutput = new EncodingOutput();
+		encodingOutput.setOutputId(output.getId());
+		encodingOutput.setOutputPath(OUTPUT_BASE_PATH + path);
+		encodingOutput.setAcl(Arrays.asList(new AclEntry(AclPermission.PUBLIC_READ)));
+
+		FMP4Muxing muxing = new FMP4Muxing();
+        MuxingStream list = new MuxingStream();
```
Comment (rkersche) [inline src/test/java/com/bitmovin/api/examples/CreateMultiCodecEncodingWithDashHls.java]: Incorrect indention

File: src/test/java/com/bitmovin/api/examples/CreateMultiCodecEncodingWithDashHls.java [hunk 11]
```diff
@@ -0,0 +1,718 @@
...
+	private TSMuxing createTSMuxing(Encoding encoding, Output output, String path, Stream stream)
+			throws BitmovinApiException, IOException, RestException, URISyntaxException, UnirestException
+	{
+		EncodingOutput encodingOutput = new EncodingOutput();
+		encodingOutput.setOutputId(output.getId());
+		encodingOutput.setOutputPath(OUTPUT_BASE_PATH + path);
+		encodingOutput.setAcl(Arrays.asList(new AclEntry(AclPermission.PUBLIC_READ)));
+
+		TSMuxing muxing = new TSMuxing();
+        MuxingStream list = new MuxingStream();
```
Comment (rkersche) [inline src/test/java/com/bitmovin/api/examples/CreateMultiCodecEncodingWithDashHls.java]: Incorrect indention

File: src/test/java/com/bitmovin/api/examples/CreateMultiCodecEncodingWithDashHls.java [hunk 12]
```diff
@@ -0,0 +1,718 @@
...
+		{
+			return ConfigType.VP9;
+		}
+	}
+
+	private static class InvalidCodecException extends Exception
+	{
+        private static final long serialVersionUID = 1L;
+
+        public InvalidCodecException(String msg)
```
Comment (rkersche) [inline src/test/java/com/bitmovin/api/examples/CreateMultiCodecEncodingWithDashHls.java]: Incorrect indention

**Review by rkersche — CHANGES_REQUESTED** [review]
Cool, please also send me some examples for DASH and HLS

**Comment by gfronza** [comment]
@rkersche Thanks! I guess I forgot to run the auto format because my Eclipse is configured properly.

Examples (HLS and DASH):
[manifests.zip](https://github.com/bitmovin/bitmovin-java/files/1567712/manifests.zip)

**Review by rkersche — APPROVED** [review]

**Pull request merged**